msm/src/serialization/witness.rs

```diff
@@ -156,46 +160,119 @@ pub fn deserialize_field_element<Fp: Field, Env: InterpreterEnv<Fp>>(
     }
 
     // FIXME: range check
-    let c0 = Env::get_column_for_msm_limb(0);
-    let c1 = Env::get_column_for_msm_limb(1);
-    let c2 = Env::get_column_for_msm_limb(2);
-    let c3 = Env::get_column_for_msm_limb(3);
-    let c4 = Env::get_column_for_msm_limb(4);
-    let c5 = Env::get_column_for_msm_limb(5);
-
-    env.bitmask_be(&input_limb0, 15, 0, c0);
-    env.bitmask_be(&input_limb0, 30, 15, c1);
-    env.bitmask_be(&input_limb0, 45, 30, c2);
-    env.bitmask_be(&input_limb0, 60, 45, c3);
-    env.bitmask_be(&input_limb0, 75, 60, c4);
-    {
+    let c0_var = {
+        let c0 = Env::get_column_for_msm_limb(0);
+        env.bitmask_be(&input_limb0, 15, 0, c0)
+    };
+    let c1_var = {
+        let c1 = Env::get_column_for_msm_limb(1);
+        env.bitmask_be(&input_limb0, 30, 15, c1)
+    };
+    let c2_var = {
+        let c2 = Env::get_column_for_msm_limb(2);
+        env.bitmask_be(&input_limb0, 45, 30, c2)
+    };
+    let c3_var = {
+        let c3 = Env::get_column_for_msm_limb(3);
+        env.bitmask_be(&input_limb0, 60, 45, c3)
+    };
+    let c4_var = {
+        let c4 = Env::get_column_for_msm_limb(4);
+        env.bitmask_be(&input_limb0, 75, 60, c4)
+    };
+    let c5_var = {
+        let c5 = Env::get_column_for_msm_limb(5);
         let res = (limbs[0] >> 75) & ((1 << (88 - 75)) - 1);
         let res_prime = limbs[1] & ((1 << 2) - 1);
         let res = res + (res_prime << (15 - 2));
-        let res = Env::constant(res.into());
-        env.copy(&res, c5);
-    }
+        let res = Env::constant(Fp::from(res));
+        env.copy(&res, c5)
+    };
 
     // Processing limbs1
     // FIXME: range check
-    let c6 = Env::get_column_for_msm_limb(6);
-    let c7 = Env::get_column_for_msm_limb(7);
-    let c8 = Env::get_column_for_msm_limb(8);
-    let c9 = Env::get_column_for_msm_limb(9);
-    let c10 = Env::get_column_for_msm_limb(10);
-    let c11 = Env::get_column_for_msm_limb(11);
-    env.bitmask_be(&input_limb1, 17, 2, c6);
-    env.bitmask_be(&input_limb1, 32, 17, c7);
-    env.bitmask_be(&input_limb1, 47, 32, c8);
-    env.bitmask_be(&input_limb1, 62, 47, c9);
-    env.bitmask_be(&input_limb1, 77, 62, c10);
-    {
+    let c6_var = {
```

Comment:
I am thinking about a better interface where you could just bitmask certain bit positions from a biguint or foreign field element instead of the limbs for less error-proneness. Specially thinking about these "middle" steps where you have to get bits from two different limbs.

Reply:
Opening https://github.com/o1-labs/proof-systems/issues/1900.
I think it must be included in a more generalized way of handing the input of the function.